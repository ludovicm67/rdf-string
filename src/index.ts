export { sparql, SparqlTemplateResult, SparqlValue } from './lib/sparql'
