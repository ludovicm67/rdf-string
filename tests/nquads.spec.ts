import RDF from '@rdfjs/dataset'
import { blankNode, literal, quad } from '@rdfjs/data-model'
import namespace from '@rdfjs/namespace'
import { schema, xsd } from '@tpluscode/rdf-ns-builders'
import { nquads } from '../src'

const ex = namespace('http://example.com/')

describe('nquads', () => {
  describe('interpolating JS types', () => {
    it('leaves strings intact', () => {
      // when
      const foo = 'foo'
      const str = nquads`<http://example.com/${foo}>`.toString()

      // then
      expect(str).toEqual('<http://example.com/foo>')
    })
  })

  describe('interpolating dataset', () => {
    it('writes all graphs', () => {
      // given
      const salary = blankNode('salary')
      const dataset = RDF.dataset()
        .add(quad(ex.John, schema.name, literal('John'), ex.John))
        .add(quad(ex.Jane, schema.name, literal('Jane'), ex.Jane))
        .add(quad(ex.Jane, schema.spouse, ex.Jane))
        .add(quad(ex.John, schema.baseSalary, salary, ex.John))
        .add(quad(salary, schema.value, literal('20000', xsd.int), ex.John))

      // when
      const str = nquads`${dataset}`.toString()

      // then
      expect(str).toMatchSnapshot()
      expect(str).toBeValidNQuads()
    })
  })
})
