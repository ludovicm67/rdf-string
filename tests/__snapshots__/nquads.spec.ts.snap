// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nquads interpolating dataset writes all graphs 1`] = `
"<http://example.com/John> <http://schema.org/name> \\"John\\" <http://example.com/John> .
<http://example.com/John> <http://schema.org/baseSalary> _:salary <http://example.com/John> .
_:salary <http://schema.org/value> \\"20000\\"^^<http://www.w3.org/2001/XMLSchema#int> <http://example.com/John> .
<http://example.com/Jane> <http://schema.org/name> \\"Jane\\" <http://example.com/Jane> .
<http://example.com/Jane> <http://schema.org/spouse> <http://example.com/Jane>  ."
`;
